rosidl_typesupport_cpp/cmake/rosidl_typesupport_cpp_generate_action_interfaces.cmake

```diff
@@ -0,0 +1,195 @@
+set(_generated_files "")
+
+foreach(_idl_file ${rosidl_generate_action_interfaces_IDL_FILES})
+  get_filename_component(_extension "${_idl_file}" EXT)
+  get_filename_component(_parent_folder "${_idl_file}" DIRECTORY)
+  get_filename_component(_parent_folder "${_parent_folder}" NAME)
+  if(_extension STREQUAL ".action")
+    set(_allowed_parent_folders "action")
+    if(NOT _parent_folder IN_LIST _allowed_parent_folders)
+      message(FATAL_ERROR "Interface file with unknown parent folder: ${_idl_file}")
+    endif()
+  else()
+    message(FATAL_ERROR "Interface file with unknown extension: ${_idl_file}")
+  endif()
+  get_filename_component(_msg_name "${_idl_file}" NAME_WE)
+  string_camel_case_to_lower_case_underscore("${_msg_name}" _header_name)
+  list(APPEND _generated_files
+    "${_output_path}/${_parent_folder}/${_header_name}__type_support.cpp"
+  )
+endforeach()
+
+set(_dependency_files "")
+set(_dependencies "")
+foreach(_pkg_name ${rosidl_generate_action_interfaces_DEPENDENCY_PACKAGE_NAMES})
+  foreach(_idl_file ${${_pkg_name}_INTERFACE_FILES})
+  get_filename_component(_extension "${_idl_file}" EXT)
+  if(_extension STREQUAL ".msg")
+    set(_abs_idl_file "${${_pkg_name}_DIR}/../${_idl_file}")
+    normalize_path(_abs_idl_file "${_abs_idl_file}")
+    list(APPEND _dependency_files "${_abs_idl_file}")
+    list(APPEND _dependencies "${_pkg_name}:${_abs_idl_file}")
+  endif()
+  endforeach()
+endforeach()
+
+set(target_dependencies
+  "${rosidl_typesupport_cpp_BIN}"
+  ${rosidl_typesupport_cpp_GENERATOR_FILES}
+  "${rosidl_typesupport_cpp_TEMPLATE_DIR}/action__type_support.cpp.em"
+  ${rosidl_generate_action_interfaces_IDL_FILES}
+  ${_dependency_files})
+foreach(dep ${target_dependencies})
+  if(NOT EXISTS "${dep}")
+    get_property(is_generated SOURCE "${dep}" PROPERTY GENERATED)
+    if(NOT ${_is_generated})
+      message(FATAL_ERROR "Target dependency '${dep}' does not exist")
+    endif()
+  endif()
+endforeach()
+
+set(generator_arguments_file "${CMAKE_CURRENT_BINARY_DIR}/rosidl_typesupport_cpp__generate_action_interfaces__arguments.json")
+rosidl_write_generator_arguments(
+  "${generator_arguments_file}"
+  PACKAGE_NAME "${PROJECT_NAME}"
+  ROS_INTERFACE_FILES "${rosidl_generate_action_interfaces_IDL_FILES}"
+  ROS_INTERFACE_DEPENDENCIES "${_dependencies}"
+  OUTPUT_DIR "${_output_path}"
+  TEMPLATE_DIR "${rosidl_typesupport_cpp_TEMPLATE_DIR}"
+  TARGET_DEPENDENCIES ${target_dependencies}
+)
+
+get_used_typesupports(typesupports "rosidl_typesupport_cpp")
+add_custom_command(
+  OUTPUT ${_generated_files}
+  COMMAND ${PYTHON_EXECUTABLE} ${rosidl_typesupport_cpp_BIN}
+  --generator-arguments-file "${generator_arguments_file}"
+  --typesupports ${typesupports}
+  DEPENDS ${target_dependencies}
+  COMMENT "Generating C++ type support dispatch for ROS action interfaces"
+  VERBATIM
+)
+
+set(_target_suffix "__generate_action_interfaces__rosidl_typesupport_cpp")
```

Comment:
@apojomovsky @sloretz any reason for this to be reversed with respect to https://github.com/ros2/rosidl_typesupport/pull/36/files#diff-d9cdf0cebb0a07a690ecd5bcec505105R89 ?

Reply:
No reason

Reply:
Made consistent in 31efacab64bd29704b624dda92e1980c356f5cbc